homeassistant/components/zamg/config_flow.py

```diff
@@ -92,7 +92,7 @@ async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
 
         return await self.async_step_user(
             user_input={
-                CONF_STATION_ID: int(config[CONF_STATION_ID]),
-                CONF_NAME: config.get(CONF_NAME, ""),
+                CONF_STATION_ID: int(station_id),
+                CONF_NAME: config.get(CONF_NAME) or station_id,
```

Comment:
I think I lost track of the intention.
In https://github.com/home-assistant/core/pull/66469/commits/bc8766c6a906e96e76ec3c9f9d73ec667667874e you handled `""` and replaced it with the station name. Now it will always be station_id and never use station name!?

Reply:
I wanted to take just the first line (95), but i took accidentally both changes ;-(
The name is now taken from the `configuration.yaml` or the reported station name from zamg.